state/filesystem_test.go

```diff
@@ -533,6 +533,24 @@ func (s *FilesystemStateSuite) TestDestroyFilesystem(c *gc.C) {
 	assertDestroy()
 }
 
+func (s *FilesystemStateSuite) TestDestroyFilesystemStorageAssigned(c *gc.C) {
+	// Create a filesystem-type storage instance, and show that we
+	// cannot destroy the filesystem while there is storage assigned.
+	_, u, storageTag := s.setupSingleStorage(c, "filesystem", "rootfs")
+	err := s.State.AssignUnit(u, state.AssignCleanEmpty)
+	c.Assert(err, jc.ErrorIsNil)
+	_, err = u.AssignedMachineId()
+	c.Assert(err, jc.ErrorIsNil)
+	filesystem := s.storageInstanceFilesystem(c, storageTag)
+
+	err = s.State.DestroyFilesystem(filesystem.FilesystemTag())
+	c.Assert(err, gc.ErrorMatches, "destroying filesystem 0/0: filesystem is assigned to storage data/0")
+
+	removeStorageInstance(c, s.State, storageTag)
+	err = s.State.DestroyFilesystem(filesystem.FilesystemTag())
```

Comment:
Ah, hard to tell from the diff. There's a local func defined on line 526. This destroys and checks life. Perhaps that func could be extracted to live on the test struct and used in multiple tests.